diff --git a/Makefile b/Makefile
index b368eff..d19aa31 100644
--- a/Makefile
+++ b/Makefile
@@ -13,14 +13,15 @@ TARGETS := libapriltag.a libapriltag.so
 
 .PHONY: all
 all: $(TARGETS)
-	@$(MAKE) -C example all
 
 .PHONY: install
 install: libapriltag.so
 	@chmod +x install.sh
+	@./install.sh $(PREFIX)/lib libapriltag.so.3.0.0
+	@./install.sh $(PREFIX)/lib libapriltag.so.3
 	@./install.sh $(PREFIX)/lib libapriltag.so
 	@./install.sh $(PREFIX)/include/apriltag $(APRILTAG_HEADERS)
-	@ldconfig
+
 
 libapriltag.a: $(APRILTAG_OBJS)
 	@echo "   [$@]"
@@ -28,7 +29,9 @@ libapriltag.a: $(APRILTAG_OBJS)
 
 libapriltag.so: $(APRILTAG_OBJS)
 	@echo "   [$@]"
-	@$(CC) -fPIC -shared -o $@ $^
+	@$(CC) -fPIC -shared -lm -lpthread -Wl,-soname,libapriltag.so.3 -o libapriltag.so.3.0.0 $^
+	ln -s libapriltag.so.3.0.0 libapriltag.so.3
+	ln -s libapriltag.so.3 libapriltag.so
 
 %.o: %.c
 	@echo "   $@"
@@ -37,4 +40,5 @@ libapriltag.so: $(APRILTAG_OBJS)
 .PHONY: clean
 clean:
 	@rm -rf *.o common/*.o $(TARGETS)
+	@rm libapriltag.so.3 libapriltag.so.3.0.0
 	@$(MAKE) -C example clean
diff --git a/install.sh b/install.sh
index 04bb5b2..158b272 100755
--- a/install.sh
+++ b/install.sh
@@ -11,6 +11,6 @@ shift
 for src in "$@"; do
     dest=$TARGETDIR/$src
     mkdir -p $(dirname $dest)
-    cp $src $dest
+    cp -d $src $dest
     echo $dest
 done
